tracee-rules/input.go

```diff
@@ -4,47 +4,49 @@ import (
 	"bufio"
 	"encoding/gob"
 	"encoding/json"
+	"errors"
 	"fmt"
 	"io"
 	"log"
 	"os"
+	"strings"
 	"time"
 
 	"github.com/aquasecurity/tracee/tracee-rules/types"
 	tracee "github.com/aquasecurity/tracee/tracee/external"
 )
 
-func setupStdinSource(inputSource string) (chan types.Event, error) {
-	res := make(chan types.Event)
-	scanner := bufio.NewScanner(os.Stdin)
-	go func() {
-		for scanner.Scan() {
-			event := scanner.Bytes()
-			switch inputSource {
-			case "tracee":
-				var e tracee.Event
-				err := json.Unmarshal(event, &e)
-				if err != nil {
-					log.Printf("invalid json in %s: %v", string(event), err)
-				}
-				res <- types.Event(e)
-			}
-		}
-		close(res)
-	}()
-	return res, nil
+var helpErr = errors.New("help me if you can im feeling down")
```

Comment:
:-)

Reply:
a better way to implement this would be to return a typed error, and check for the error type, instead of it's string representation.

Reply:
I did this based on how the Go standard library does it (i.e. io.EOF in https://golang.org/pkg/io/#pkg-variables). 

Do you think it's necessary to create a new type? We'd then have to have a type switch on return and it'd be effectively doing the same thing.

Reply:
I don't think that's a comparable example, but it's not that important, this is fine as well